go/vt/vtgate/planbuilder/operators/hash_join.go

```diff
@@ -69,14 +82,15 @@ func NewHashJoin(lhs, rhs Operator, outerJoin bool) *HashJoin {
 		LHS:      lhs,
 		RHS:      rhs,
 		LeftJoin: outerJoin,
+		columns:  &hashJoinColumns{},
 	}
 	return hj
 }
 
 func (hj *HashJoin) Clone(inputs []Operator) Operator {
 	kopy := *hj
 	kopy.LHS, kopy.RHS = inputs[0], inputs[1]
-	kopy.columns = slices.Clone(hj.columns)
+	kopy.columns = &hashJoinColumns{columns: slices.Clone(hj.columns.columns)}
 	kopy.LHSKeys = slices.Clone(hj.LHSKeys)
 	kopy.RHSKeys = slices.Clone(hj.RHSKeys)
 	return &kopy
```

Comment:
We don't seem to be cloning the `HashJoin.JoinComparisons` field, is that okay?

Reply:
no it is not! good catch